src/components/nativestorage_driver.coffee

```diff
@@ -25,15 +25,18 @@ _initStorage = (options) ->
         .catch (error) ->
           _setKeys([], dbInfo)
 
+_patchPromise = (promise) ->
+  promise.catch ?= promise.fail
```

Comment:
Btw I would prefer if these changes go into a separate PR as they are unrelated to the structure of this lib.